akka-actor/src/main/scala/akka/io/Dns.scala

```diff
@@ -89,3 +89,11 @@ class DnsExt(system: ExtendedActorSystem) extends IO.Extension {
 
   def getResolver: ActorRef = manager
 }
+
+object IpVersionSelector {
+  def getInetAddress(ipv4: Option[Inet4Address], ipv6: Option[Inet6Address]): Option[InetAddress] =
+    sys.props.get("java.net.preferIPv6Addresses") match {
```

Comment:
Thanks @johanandren for the tip, I did not know about that possibility. Looks a bit cumbersome for the task.

I'm agree with @patriknw. As he pointed, the logic is trivial furthermore it is the behaviour that anyone should expected using the "java.net.preferIPv6Addresses" property. Currently this scenario is not common but it will be start to be soon.

As I commented before, I did not use a `val` due to the properties nature. If the Akka team want, it can be easily turned into `val` as the switch of this property on hot is very rare but you should be aware that another bug can be raised in the future about that.